components/browser/toolbar/src/main/java/mozilla/components/browser/toolbar/BrowserToolbar.kt

```diff
@@ -219,7 +219,14 @@ class BrowserToolbar @JvmOverloads constructor(
         get() = displayToolbar.titleView.textSize
         set(value) {
             val newSize = minOf(value, MAX_TITLE_SIZE)
-            displayToolbar.titleView.textSize = newSize
+            // We only want the text size to scale smaller (with sp) if it's below the max size
+            val unitType = if (newSize == MAX_TITLE_SIZE) {
+                TypedValue.COMPLEX_UNIT_DIP
+            } else {
+                TypedValue.COMPLEX_UNIT_SP
+            }
```

Comment:
Something is not right here (even though the result may look okay on your device).

First we compare values (`minOf`), then later on we use different units -> The comparison isn't comparing the same thing. This seems wrong. Do we need to have both values converted to actual pixels before we can compare? Or am I missing something?